pkg/app/component_test.go

```diff
@@ -498,10 +519,15 @@ type bar struct {
 	appInstalled bool
 	appRezized   bool
 	updated      bool
+	title        string
+}
+
+func (b *bar) OnInit() {
+	b.title+="bar"
 }
 
 func (b *bar) OnPreRender(ctx Context) {
-	ctx.Page().SetTitle("bar")
+	ctx.Page().SetTitle(b.title)
```

Comment:
We want to test whether the event was called. This test if a page title was set.
That also brings the question of whether `OnInit` should also be called on the server side when a component is pre-rendered.

Reply:
It has to be called because it should always be called before the component is rendered. This is also why I test for both, server and client rendering. It won't be called if it is not being used. If it is being used in Prerender and Render it most probably also needs the init with both. If not one could check for server and behave differently.

Reply:
See my comment about the title field